frontend/src/components/WorkspaceChat/ChatContainer/ChatHistory/Citation/index.jsx

```diff
@@ -122,7 +122,7 @@ function CitationDetailModal({ source, onClose }) {
         <div className="flex items-start justify-between p-6 border-b rounded-t border-gray-500/50">
           <div className="flex flex-col flex-grow mr-4">
             <h3 className="text-xl font-semibold text-white overflow-hidden overflow-ellipsis whitespace-nowrap">
-              {truncate(title, 52)}
+              {truncate(title, 45)}
```

Comment:
We should truncate shorter since that does appear to be an issue, however this does not work for all screen sizes. If possible we should _also_ be absolutely positioning the exit button since that will ensure even if we overflow on small screens - the button is at least still present.